src/Scripting/ImGuiBindings.cpp

```diff
@@ -0,0 +1,527 @@
+#include "ImGuiBindings.h"
+
+#include <imgui/imgui.h>
+#include <string>
+#include <vector>
+#include <tuple>
+#include <sol/sol.hpp>
+
+// Windows
+bool imGuiBegin(const std::string &name) { return ImGui::Begin(name.c_str()); }
+std::tuple<bool, bool> imGuiBegin(const std::string &name, bool isOpened, int flags) {
+    bool shouldDraw = ImGui::Begin(name.c_str(), &isOpened, static_cast<ImGuiWindowFlags_>(flags));
+    return { shouldDraw, isOpened };
+}
+void End() { ImGui::End(); }
+
+// Child Windows
+bool imGuiBeginChild(const std::string &name, float sizeX, float sizeY, bool border) { return ImGui::BeginChild(name.c_str(), { sizeX, sizeY }, border); }
+void imGuiEndChild() { ImGui::EndChild(); }
+
+// Windows Utilities
+bool imGuiIsWindowHovered(int flags) { return ImGui::IsWindowHovered(static_cast<ImGuiHoveredFlags>(flags)); }
```

Comment:
Can we deal away with `static_cast`s here and in other binding functions and pass in the `enum`s directly? Is it supported by sol2?